File: PHP/Laptop.php
<?php
require_once "Product.php"; // Import Product class

class Laptop extends Product {
    // Additional properties specific to Laptop
    private string $processor; // Laptop processor type
    private int $ram;          // RAM capacity (in GB)
    private int $battery;      // Battery capacity (in mAh)
    private int $warranty;     // Warranty period (in years)

    // Constructor with default values
    public function __construct(
        int $id = 0,
        string $name = "",
        string $brand = "",
        int $price = 0,
        string $processor = "",
        int $ram = 0,
        int $battery = 0,
        int $warranty = 0,
        string $image = ""
    ) {
        // Call parent constructor (Product)
        parent::__construct($id, $name, $brand, $price, $image);

        // Initialize Laptop-specific properties
        $this->processor = $processor;
        $this->ram = $ram;
        $this->battery = $battery;
        $this->warranty = $warranty;
    }

    // Getters (to access private properties)
    public function getProcessor(): string {
        return $this->processor;
    }

    public function getRam(): int {
        return $this->ram;
    }

    public function getBattery(): int {
        return $this->battery;
    }

    public function getWarranty(): int {
        return $this->warranty;
    }

    // Setters (to modify private properties)
    public function setProcessor(string $processor): void {
        $this->processor = $processor;
    }

    public function setRam(int $ram): void {
        $this->ram = $ram;
    }

    public function setBattery(int $battery): void {
        $this->battery = $battery;
    }

    public function setWarranty(int $warranty): void {
        $this->warranty = $warranty;
    }
}

File: Cpp/Laptop.cpp
#include "Product.cpp"

class Laptop : public Product {
    private:
        string processor;  // Processor type (e.g., Intel i7, AMD Ryzen)
        int ram;           // RAM size in GB
        int battery;       // Battery capacity inn mAh
        int warranty;      // Warranty period in years
    
    public:
        // Default constructor
        Laptop() : Product() {
            this->processor = "";
            this->ram = 0;
            this->battery = 0;
            this->warranty = 0;
        }

        // Parameterized constructor
        Laptop(int id, string name, string brand, int price, string processor, int ram, int battery, int warranty) 
            : Product(id, name, brand, price) {
            this->processor = processor;
            this->ram = ram;
            this->battery = battery;
            this->warranty = warranty;
        }

        // Getters
        string getProcessor() {
            return this->processor;
        }

        int getRam() {
            return this->ram;
        }

        int getBattery() {
            return this->battery;
        }

        int getWarranty() {
            return this->warranty;
        }

        // Setters
        void setProcessor(string processor) {
            this->processor = processor;
        }

        void setRam(int ram) {
            this->ram = ram;
        }

        void setBattery(int battery) {
            this->battery = battery;
        }

        void setWarranty(int warranty) {
            this->warranty = warranty;
        }

        // Destructor
        ~Laptop() {
        }
};